src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -26,20 +39,21 @@ const default_output_attributes = Dict(
 
 An output writer for writing to NetCDF files.
 """
-mutable struct NetCDFOutputWriter{D, O, I, T, S} <: AbstractOutputWriter
-              filename :: String
+mutable struct NetCDFOutputWriter{D, O, I, T, S, A} <: AbstractOutputWriter
+              filepath :: String
                dataset :: D
                outputs :: O
     iteration_interval :: I
          time_interval :: T
                   mode :: String
-                slices :: S
+          field_slicer :: S
+            array_type :: A
               previous :: Float64
                verbose :: Bool
 end
 
 """
-    NetCDFOutputWriter(model, outputs; filename, iteration_interval=nothing, time_interval=nothing,
+    NetCDFOutputWriter(model, outputs; filepath, iteration_interval=nothing, time_interval=nothing,
```

Comment:
In `JLD2OutputWriter` we have two arguments: `dir`, and `filename`. This is combined internally into a `filepath`. I suppose the reasoning is that its common to have multiple output writers (which you'd want to save data into the same directory), but different names. Having two keyword arguments makes that slightly more convenient. I'm ok with using `filepath` instead if there's a good argument. But I think both output writers should have the same interface?

Reply:
Actually, looks like `JLD2OutputWriter` uses "prefix":

https://github.com/CliMA/Oceananigans.jl/blob/2281da8a6c1857024c38c622f2729fb760645f61/src/OutputWriters/jld2_output_writer.jl#L30

The distinction is that prefix is appended with various endings (eg `.jld2`, or sometimes `_part$n.jld2`) to complete the file name.

Happy to change `JLD2OutputWriter` to conform to a style we think is clearest --- let's discuss.

Reply:
I actually think a friendly interface would use `filename`, but allow the user to give only the first "part" of the file name, leaving out `.jld2` or `.nc` --- which are already made obvious by the name of the object (`NetCDF*` or `JLD2*`). This has the advantage that this keyword argument doesn't need to be changed if switching between output writers.

Reply:
Yeah I agree a shared interface would be good, but I think the reason `JLD2OutputWriter` has both a `dir` and `filename` is to support file splitting with automatic file naming.

The `filepath` should include both the directory and filename so I think having two arguments is kind of clunky. Ideally both would just use a `filepath`.

I guess `NetCDFOutputWriter` gets away with just a `filepath` because it doesn't support file splitting so the filename never changes during the course of a simulation.

We could modify the `JLD2OutputWriter` to just take in a `filepath`. Two possible solutions are:
1. File splitting could be supported by injecting `part1`, `part2`, etc. appropriately into the `filepath`, or
2. via another kwarg `filename_pattern` that's a function `filename_pattern(n::Int)::String` that returns the filename for part `n`.

Reply:
A third alternative would be to remove the ability to split files, but I think this is a bad idea. While there is no file size limit for HDF5 and we typically don't split files since our files aren't huge, there will be users in the future who will run huge models and need this functionality due to memory limits (or even filesystem limits).

Reply:
> I guess NetCDFOutputWriter gets away with just a filepath because it doesn't support file splitting so the filename never changes during the course of a simulation.

Do we plan on supporting file splitting for `NetCDFOutputWriter` in the future?

Reply:
Yeah I think we should. We should continue discussion in #884 (where it seems we already liked `filepath` haha).